### ISSUE TITLE
Facilitate incorporation of user-specified loss functions

### ISSUE THREAD

**Issue by eigenvivek** [eigenvivek/xvr#31]
The image similarity loss function for registration is currently hardcoded:

https://github.com/eigenvivek/xvr/blob/f31fb93c14d9ef5867f06309eba23857c426545e/src/xvr/registrar.py#L82

Would be good to redesign the loss function such that a user can define their own loss function.

**Comment by ryukyungjoon** [comment]
Hello, Vivek
It is mean that similarity measure can be change another measure? like ncc, ssim etc.
it is up to me. right?

**Comment by eigenvivek** [comment]
Hi @ryukyungjoon , no you can't change the image similarity measure in the current implementation of `xvr`.

The current loss function is a convex combination of Multiscale NCC and Gradient NCC (you can set the relative importance each metric with `beta`):
https://github.com/eigenvivek/xvr/blob/f31fb93c14d9ef5867f06309eba23857c426545e/src/xvr/registrar.py#L79-L82

In the future, we will support more image similarity metrics (and perhaps user-defined metrics, too).

**Comment by eigenvivek** [comment]
In the dev branch, users can now pass a `torch.nn.Module` whose forward method can be used as a loss function:
https://github.com/eigenvivek/xvr/blob/c102c5463bd4ccc72dfbd27e69797ef82242110c/src/xvr/register/base.py#L43
